src/Microsoft.AspNet.Routing/Infrastructure/PathSegment.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Routing.Infrastructure
```

Comment:
`Internal` it is for `PathTokenizer`. Per discussion with @davidfowl - moving `PathSegment` + `StringSegment` to primitives.